### PR TITLE
bugfix sensorless not returning to run_current after homing

### PR DESCRIPTION
**Pull request by rogerlz**
bugfix sensorless not returning to run_current after homing

I also took the opportunity to start a new BaseTMCCurrentHelper() class to reduce the duplicated code and added comments to the code to help people working on it in the future.

### CODE REVIEW

File: klippy/extras/tmc.py [hunk 1]
```diff
@@ -744,3 +749,76 @@ def TMCStealthchopHelper(config, mcu_tmc, tmc_freq):
...
+            maxval=max_current,
+        )
+        self.current_change_dwell_time = config.getfloat(
+            "current_change_dwell_time", 0.5, above=0.0
+        )
+        self.req_run_current = self.config_run_current
+        self.req_hold_current = self.config_hold_current
+        self.req_home_current = self.config_home_current
+
+        self.actual_current = self.req_run_current
```
Comment (bwnance) [inline klippy/extras/tmc.py:779]: **nitpick:** let's add some inline comments around what these are all for and what stuff stands for (req = requested, actual = actual current running on driver, etc)

File: klippy/extras/tmc5160.py [hunk 2]
```diff
@@ -250,42 +250,22 @@
...
-            "home_current", self.run_current, above=0.0, maxval=MAX_CURRENT
-        )
-        self.current_change_dwell_time = config.getfloat(
-            "current_change_dwell_time", 0.5, above=0.0
-        )
-        self._prev_current = self.run_current
-        self.req_hold_current = self.hold_current
+        super().__init__(config, mcu_tmc, MAX_CURRENT)
+
+        self.actual_current = self.req_run_current
```
Comment (bwnance) [inline klippy/extras/tmc5160.py:257]: i think this is superfluous now? looks like the same line gets run in the base

File: klippy/extras/tmc.py [hunk 3]
```diff
@@ -744,3 +749,76 @@ def TMCStealthchopHelper(config, mcu_tmc, tmc_freq):
     else:
         # TMC2208 uses en_spreadCycle
         fields.set_field("en_spreadcycle", not en_pwm_mode)
+
+
+class TMCCurrentHelper:
```
Comment (bwnance) [inline klippy/extras/tmc.py:754]: let's rename to `BaseTMCCurrentHelper`

**Review by bwnance — CHANGES_REQUESTED** [review]

File: klippy/extras/tmc5160.py (hunk 2 above)
Comment (rogerlz) [reply klippy/extras/tmc5160.py]: yes it is.. did not notice I forgot it there

**Review by bwnance — DISMISSED** [review]
i like this - i might take a swing at a reorg of this down the road but this is more than enough for now :)

**Review by bwnance — APPROVED** [review]

**Pull request merged**